### PR TITLE
Add sgn() function to the math module

### PR DESCRIPTION
**Pull request by johnnovak**

### CODE REVIEW

**Comment by data-man** [comment]
Why not sign?
For uint* would also be useful when sgn will be used in macros and templates.

**Comment by johnnovak** [comment]
Usually the function is called `sgn` in maths (https://en.wikipedia.org/wiki/Sign_function), it's kind of a standard name.

Good point on adding uint types, I'll do that now.

File: lib/pure/math.nim [hunk 1]
```diff
@@ -395,6 +395,12 @@ proc radToDeg*[T: float32|float64](d: T): T {.inline.} =
   ## Convert from radians to degrees
   result = T(d) / RadPerDeg
 
+proc sgn*[T: SomeNumber](x: T): int {.inline.} =
+  ## Sign function. Returns -1 for negative numbers and `NegInf`, 1 for
+  ## positive numbers and `Inf`, and 0 for positive zero, negative zero and
+  ## `NaN`.
+  cast[int](T(0) < x) - cast[int](x < T(0))
```
Comment (Araq) [inline lib/pure/math.nim]: Instead of ``cast[int]`` use ``ord(x)`` here.

**Comment by yglukhov** [comment]
Why would you miss `i` in the function name?
GLSL, HLSL, Javascript: "sign"
Python: "copysign"

**Comment by yglukhov** [comment]
Julia: sign, copysign
C#: Sign
Rust, Haskel, Lisp: signum
Java: copysign, signum
Crystal: copysign

Ok, I admit, there's D: sgn =)

**Comment by Araq** [comment]
I ``sign`` my invoices though, so ``sgn`` is a better name, IMO.

File: lib/pure/math.nim (hunk 1 above)
Comment (johnnovak) [inline lib/pure/math.nim]: Good point, I'll change it.

**Comment by johnnovak** [comment]
@yglukhov To be amicably argumentative, if we cared too much about what everybody else is doing then Nim would be another variant of Java, C#, Scala or whatever... :)

I like `sgn` a bit better because that's the mathematical convention and as @Araq hinted, it's immediately obvious that it's the math function you're using when you see `sgn(x)` in the code. And it's shorter, that's why we have `sin`, not `sine`, `atan` and not `arctan` or `arctangent` etc.

But I can live with `sign` too, I can always create an alias for it in my code :)

**Comment by konqoro** [comment]
System has the following non-concrete type: `SomeNumber = SomeInteger | SomeReal` so maybe change the generic this?

**Comment by johnnovak** [comment]
@konqoro Yeah, why not :) Done.

**Comment by dom96** [comment]
I vote for ``sgn`` too. Mainly because it's unambiguous: [WP:Sign](https://en.wikipedia.org/wiki/Sign) vs. [WP:Sgn](https://en.wikipedia.org/wiki/Sgn) :)

**Pull request merged**